export.js

```diff
@@ -0,0 +1,161 @@
+function grabExcelValues(rows, artifact, objTemplate, customFieldRange) {
+    if (rows === null) {
+        getRows(artifact, objTemplate, customFieldRange);
+    }
+    else {
+        return Excel.run(function (context) {
+            let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+            let inputRange = columnRanges[artifact] + rows;
+            let customRange = (customFieldRange[0] + 3) + ":" + (customFieldRange[1] + rows);
+            let inputValues = sheet.getRange(inputRange);
+            let customFields = sheet.getRange(customRange);
+            inputValues.load();
+            customFields.load();
+            return context.sync()
+                .then(function () {
+                    customFieldObjArr = customFields.values.map(customFieldObjCreate);
+                    ajaxExport(inputValues.values, artifact, objTemplate, customFieldObjArr);
+                })
+        });
+    }
+}
+
+function getRows(artifact, objTemplate, customFieldRange) {
+    return Excel.run(function (context) {
+        let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+        let sheetRange = sheet.getUsedRange();
+        sheetRange.load();
+        return context.sync()
+            .then(function () {
+                grabExcelValues(sheetRange.values.length, artifact, objTemplate, customFieldRange);
+            })
+            .catch(function (error) {
+                console.log(error);
+            })
+    });
+}
+
+function ajaxExport(valueArray, artifact, objTemplate, customFieldObjArr) {
```

Comment:
so this function is passed in general data about the main fields which is built into the right format for the array, and is passed in the cleansed custom field data? Is there a reason why these two are built at different stages?